tesla-hero-slider/src/components/slider-navigation.js

```diff
@@ -0,0 +1,39 @@
+import React, {Component} from 'react';
+import propTypes from 'prop-types';
+import './../styles/scss/components/slider-navigation.css';
+
+class SliderNavigation extends Component {
+
+	static propTypes = {
+		setActiveSlide: propTypes.func.isRequired,
+		carsNames: propTypes.array.isRequired
+	}
+
+	changeActiveSlide(e) {
+		e.preventDefault();
+		this.props.setActiveSlide(e.target.getAttribute('data-id'));
+	}
+
+	render() {
+		return (
+			<div className='tesla-slider-navigation'>
+				<ul className='tesla-slider-navigation__list'>
+					{this.props.carsNames.map((car) => {
+						return (
+							<li key={car.id} className='tesla-slider-navigation__item'>
+								<a href='#' onClick={this.changeActiveSlide.bind(this)}
```

Comment:
Avoid bind inside a render function - usually, it means it's something wrong with the code. Bind returns a new function so if you're doing it inside a render fn, it doesn't give any performance/memory boost. From the dev perspective, it can be confusing and someone can expect to have the same function every time. In this case, I'd convert it to simple lambda fn:

```js
onClick={event => {
  event.preventDefault();
  this.props.setActiveSlide(this.props.carsNames.indexOf(car);
}}
```

This also solves another problem: `data-id`. React is all about the state so the state should be encapsulated in your app. By using data attributes state is leaking into the DOM, which is something sometimes useful (for example when another app needs it). But when you're using DOM to pass pieces of information between different parts of the same app it means something went wrong :)